### ISSUE TITLE
Define naming convention for Flyway migration scripts 

### ISSUE THREAD

**Issue by wslulciuc** [MarquezProject/marquez#237]
Let's define a naming convention for our `.sql` migration scripts under the folder `db/migration` ([others](https://stackoverflow.com/questions/9497035/how-to-use-flyway-when-working-with-feature-branches/34599349#34599349) also think this is important). I propose the following format:

```
V[timestamp]__[description].sql
```

where the prefix **`V`** is followed by **`[timestamp]`** in the form **yyyyMMdd.hhmm** and **`[description]`** is a brief description of the schema change. For example:

```
V20190105.1206__add_heartbeats_table.sql
```

Thoughts?

### Resources

- [Lessons Learned Using Flyway DB with Distributed Version Control](http://www.jeremyjarrell.com/using-flyway-db-with-distributed-version-control)
- [Database Versioning and Delivery with Upgrade Scripts](https://www.infoq.com/articles/db-versioning-scripts)
- [Flyway Migrations](https://flywaydb.org/documentation/migrations)

/cc @ashulmanWeWork @hougs @sshah-wework

**Comment by hougs** [comment]
What timestamp should the author of a migration use? The time the create the file, the time the create a PR, the (aprox) time it is merged into the mainline branch? As long as we have a simpel automated way to create the ts +1.

**Comment by ashulmanWeWork** [comment]
All for the description.
For the timestamp, it's not clear to me what it gives us that the existing numerical ordering doesn't? If someone needs information about when the change was made, we can always consult GH.

**Comment by wslulciuc** [comment]
@ashulmanWeWork: 

> For the timestamp, it's not clear to me what it gives us that the existing numerical ordering doesn't?

The problem we are solving with using a timestamp for versioning instead of a incrementing a number is:

> [Incrementing a number] works great when everyone is on the same branch of code.  However, once members of the team begin working on their own branches, the likelihood of a prefix collision increases dramatically.

see: http://www.jeremyjarrell.com/using-flyway-db-with-distributed-version-control

**Comment by lulciuca** [comment]
Wouldn't conflicting versions be taken care of in a merge conflict?

**Comment by wslulciuc** [comment]
@lulciuca : Not necessarily. Let's say we had the following migrations scripts soon-to-be introduced:

*  `V2__drop_column.sql`
*  `V2__add_table.sql`

Since they're being worked on feature branches, merging them into the `master` branch wouldn't result in a merge conflict. It's only after the merge and the migration is applied (integration test would also catch this) would see errors.

**Comment by roaraya8** [comment]
I feel that by using an autogenerated date as the version it would be actually more prone to errors.
In the scenario where there are long running branches/PRs, those would contain a migration version that's in the past and might be harder to track what migrations where applied in which order.

By manually choosing the sequential version it's easier to check what was added and sort out migration conflicts when merging a branch.
That said I think we would benefit from having a script checking if there are version conflicts at build time, in case there is an oversight while merging branches.

**Comment by sshah-wework** [comment]
Schema changes happen so rarely and the number of files involved is so few that any collision on the same "V*" file name can usually be resolved by a simple file rename. 

@wslulciuc if it's ok with you, I'm going to remove this from the `0.2.0` milestone. Seems like more of a process improvement thing, but the current system is working fine.

**Comment by wslulciuc** [comment]
Yeah, that’s fine! We can pick up the discussion when we revisit DB schema changes and cleanup columns that are no longer used / relavent

**Comment by julienledem** [comment]
We use the default flyway convention: V#__
